### DISCUSSION TITLE
[Mobile] after inserting the image, the cursor should appear after the image instead of before it

### DISCUSSION THREAD

**Discussion by renntbenrennt** [singerdmx/flutter-quill#403] [Mobile]
My issue is about [Mobile]

I have tried running `example` directory successfully before creating an issue here. 

As stated in the title, and you can also see in this recorded video:

https://user-images.githubusercontent.com/13610283/130929124-b4d9922d-1bc4-46ca-889a-99894a765418.MP4

If the cursor placed before the image, it will make it really hard to create a new line for more content because it's really hard to place the cursor after the image inserted...

Thank you.

**Comment by singerdmx** [comment]
Code for this logic is at https://github.com/singerdmx/flutter-quill/commit/dc66cf13fd3732387947c67ab9b92a67bfcc4863

Flutter somehow has no way to tell the width of the image and hence we cannot determine the cursor position.

You can give it a try by playing the code around https://github.com/singerdmx/flutter-quill/commit/dc66cf13fd3732387947c67ab9b92a67bfcc4863

**Comment by renntbenrennt** [comment]
@singerdmx 

Yes, I see.

But what confused me was, from the commit message, I assumed that commit was intended to fix the problem I was talking about in this issue.

And obviously, it didn't work.(The code I am running is the latest, containing the change in the commit [you linked](https://github.com/singerdmx/flutter-quill/commit/dc66cf13fd3732387947c67ab9b92a67bfcc4863))

And now I have another idea to resolve this issue(that it will difficult to add new content after the image is inserted). But I tested it in my machine that's not working as expected.

So, the basic idea is, instead of move the cursor after the image after it's inserted, we could just programmatically add a new line after the image, that way, the cursor should be below the image and the user can do whatever they want after image insertion.

Following this assumption, I have tried add this line of code, for adding new line, 
```
controller.document.insert(length, '\n');
```
after this line of code:

https://github.com/singerdmx/flutter-quill/blob/d401879789d1bbe3cfdd6b161a1cb3e7c4b9d9a7/lib/src/widgets/toolbar/image_video_utils.dart#L72

which I assume that it's for adding image to the editor.

Sadly, it's not working as expected... maybe it's not how a newline is added or I made a mistake of passing the parameter.

**Comment by singerdmx** [comment]
Try flutter_quill: ^2.0.3, see if it still appears

**Comment by singerdmx** [comment]
If you click on the area behind the image, the cursor will be behind it
